toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -183,18 +237,34 @@ func (im *IsoMaker) setUpIsoGrub2Bootloader() {
 		fmt.Sprintf("count=%d", numberOfBlocksToCopy), // Number of blocks to copy to the output file.
 	}
 	logger.Log.Debugf("Creating an empty '%s' file of %d bytes.", im.efiBootImgPath, blockSizeInBytes*numberOfBlocksToCopy)
-	shell.MustExecuteLive("dd", ddArgs...)
+	err = shell.ExecuteLive(false /*squashErrors*/, "dd", ddArgs...)
+	if err != nil {
+		return err
+	}
 
 	logger.Log.Debugf("Formatting '%s' as an MS-DOS filesystem.", im.efiBootImgPath)
-	shell.MustExecuteLive("mkdosfs", im.efiBootImgPath)
+	err = shell.ExecuteLive(false /*squashErrors*/, "mkdosfs", im.efiBootImgPath)
+	if err != nil {
+		return err
+	}
 
 	efiBootImgTempMountDir := filepath.Join(im.buildDirPath, "efiboot_temp")
 	logger.Log.Tracef("Creating temporary mount directory '%s'.", efiBootImgTempMountDir)
-	logger.PanicOnError(os.Mkdir(efiBootImgTempMountDir, os.ModePerm))
+	err = os.Mkdir(efiBootImgTempMountDir, os.ModePerm)
+	if err != nil {
+		return err
+	}
 
 	defer func() {
 		logger.Log.Debugf("Removing '%s'.", efiBootImgTempMountDir)
-		logger.PanicOnError(os.RemoveAll(efiBootImgTempMountDir), "Failed to remove temporary mount directory '%s'.", efiBootImgTempMountDir)
+		cleanupErr := os.RemoveAll(efiBootImgTempMountDir)
+		if cleanupErr != nil {
+			if err != nil {
+				err = fmt.Errorf("failed to remove temporary mount directory '%s':\n%w\n%w", efiBootImgTempMountDir, err, cleanupErr)
```

Comment:
Similar ask here to add a word or two to separate the cause of the error in `err` from `cleanupErr`.

Reply:
updated.